dan_layer/core/src/workers/hotstuff_waiter.rs

```diff
@@ -1064,6 +1064,14 @@ where
     }
 
     fn execute(&self, payload: TPayload, object_pledges: &[ShardPledge]) -> Result<FinalizeResult, HotStuffError> {
+        let maybe_finalize_result = self
```

Comment:
maybe we need to have a hash of the pledges used, to check that it's still the same

Reply:
The payload id should have the pledges in it (we'll need to validate the transaction hash as part of mempool validation) - maybe that's all the security needed?

Reply:
The actual data in the pledges might not be correct. This could be a way for a malicious node to invoke a different result earlier

Reply:
Good point, will try work out the best way to do that check in this PR